src/io.jl

```diff
@@ -138,7 +138,17 @@ function create_internal_structures(table_tree::TableTree, connection)
         row in TulipaIO.get_table(Val(:raw), connection, "rep_periods_data")
     ]
 
-    timeframe = Timeframe(maximum(table_tree.periods.mapping.period), table_tree.periods.mapping)
+    # Calculate the total number of periods and then pipe into a Dataframe to get the first value of the df with the num_periods
+    num_periods =
+        DBInterface.execute(
+            connection,
+            "SELECT MAX(period) AS period
+                FROM rep_periods_mapping",
+        ) |>
+        DataFrame |>
+        df -> df.period |> only
+
+    timeframe = Timeframe(num_periods, TulipaIO.get_table(connection, "rep_periods_mapping"))
```

Comment:
```suggestion
    num_periods, = DuckDB.query(connection, "SELECT MAX(period) AS period FROM rep_periods_mapping")

    timeframe = Timeframe(num_periods.period, TulipaIO.get_table(connection, "rep_periods_mapping"))
```
Is this better?

You could also not rename it with `AS ...`, and instead use `num_periods[1]`.